# Task 00: Rules and Guidelines

**Read the following rules and tutorials.**

## 1. Some Rules

<div class="alert-warning">
    WARNING: If you just write down numbers without explicitly calculating them in Python, we will not give any points for that!
</div>

<p>

* **Do not use external spreadsheet programs or pocket calculators!**
    <br>**Use Python code cell only** to perform your calculations in this Jupyter notebook!<br>
* Use **formatted printing** to report the **results** of your calculations including **units!**
* **Use markdown cells** to explain your approach and answers when necessary. Discuss and reflect, when appropriate. Are your results plausible?
* **Add more code and markdown cells, if necessary.**  
* If you use information from the **internet** you will have to provide the **references** (e.g. links) to the information sources uniquely. 

## 2. Example of a Comprehensive Question-Calculation-Answer Block

**We expect you to produce an appropriate flow of question, calculation, and answer!** <br>
Appropriate means that all of your steps, thoughts, and calculations are clear and **easy to understand**. In order to grade your work, we must be able to easily follow what you are doing. If you do your work in a way that is difficult for us to understand, you will **not get a good grade**. All your calculations have to be **explicit!** If you just write down a number you calculated with a pocket calculator or a spreadsheet program we will **deduct points** for this. 

In many cases it is enough to use **formatted printing** such as `print(f"The mass of the ice block is m = {m:.2f}kg")` in code cells to report your results. **But** sometimes you have to explain and justify your approach or you have to list your references clearly. **In these case you have to use markdown cells, too!** 

### Example

**Question:** Find the desity $\rho$ of water ice in the internet. Provide the link to your source of information. What is the weight force of $2\mathrm{L}$ of ice, if its density is $\rho$ and the gravitation acceleration is $9.8\mathrm{m/s^2}$? Use SI units!

In [1]:
# Calculations:

# Given
g   = 9.8  # in m/s^2
V   = 2.0*1e-3  # in m^3
rho = 0.917*1e-3/(1e-2)**3  # in kg/m^3

m = rho*V # mass of the ice in mg
F = m*g # force in N

print(f"The mass of the ice block is m = {m:.2f}kg")
print(f"The weight force of the ice block is F = {F:.2f}N")

The mass of the ice block is m = 1.83kg
The weight force of the ice block is F = 17.97N


**Answer:** <br>
According to the Wikipedia article on [**Ice**](https://en.wikipedia.org/wiki/Ice) its density is approximately $\rho = 0.917\,\mathrm{kg/L}= 917\,\mathrm{kg/m^3}$. 
<br>_(This markdown cell is necessary to explain the chosen value and to refer to the information source)._

The weight force of the ice block is $F = 17.97\,\mathrm{kg\,m\,s^{-2}}$ (Alternatives: $F = 17.97\,\mathrm{\frac{kg\,m}{s^2}}$ or even $F = 17.97\,\mathrm{N}$).
<br>_(This markdown cell is used in addition to formatted printing. It allows more freedom in typesetting.)_